keras2onnx/common/onnx_ops.py

```diff
@@ -3,427 +3,5 @@
-
-            # Create a new name for next Tile
-            name = scope.get_unique_operator_name('Tile')
-
-        # Use the last Tile name for the name of an Identity
-        container.add_node('Identity', intermediate_output_name, output_name, op_version=1, name=name)
-    else:
-        # ONNX-1.2 has a new Tile and we use it here
-        repeat_tensor_name = scope.get_unique_variable_name(name + '_repeats')
-        container.add_initializer(repeat_tensor_name, onnx_proto.TensorProto.INT64, [len(repeats)], repeats)
-        container.add_node('Tile', [input_name, repeat_tensor_name], output_name, op_version=7, name=name)
-
-def apply_transpose(scope, input_name, output_name, container, operator_name=None, perm=None):
-    name = _create_name_or_use_existing_one(scope, 'Transpose', operator_name)
-    container.add_node('Transpose', input_name, output_name, name=name, perm=perm)
-
-def apply_upsample(scope, input_name, output_name, container, operator_name=None, mode='nearest', scales=None):
-    '''
-    :param mode: nearest or linear
-    :param scales: an integer list of scaling-up rate of all input dimensions
-    '''
-    if container.target_opset < 10:
-        name = _create_name_or_use_existing_one(scope, 'Upsample', operator_name)
-        inputs = input_name
-        attrs = {'name': name}
-        if container.target_opset < 7:
-            if len(scales) != 4:
-                raise ValueError('Need to specify a 4-element list the the scales of N-, C-, H-, and W-axes')
-            attrs['height_scale'] = float(scales[2])
-            attrs['width_scale'] = float(scales[3])
-            attrs['mode'] = mode.upper()
-            op_version = 1
-        else:
-            attrs['mode'] = mode.lower()
-            if container.target_opset < 9:
-                attrs['scales'] = list(map(float, scales))
-                op_version = 7
-            else:
-                # scales moved from attribute to input in opset 9
-                scales_tensor_name = scope.get_unique_variable_name(name + '_scales')
-                container.add_initializer(scales_tensor_name, onnx_proto.TensorProto.FLOAT, [len(scales)], scales)
-                inputs = [input_name, scales_tensor_name]
-                op_version = 9
-
-        container.add_node('Upsample', inputs, output_name, op_version=op_version, **attrs)
-    else:
-        # TODO, we need verify this after onnx opset 10 release
-        name = _create_name_or_use_existing_one(scope, 'Resize', operator_name)
-        attrs = {'name': name}
-        attrs['mode'] = mode.lower()
-
-        scales_tensor_name = scope.get_unique_variable_name(name + '_scales')
-        container.add_initializer(scales_tensor_name, onnx_proto.TensorProto.FLOAT, [len(scales)], scales)
-        inputs = [input_name, scales_tensor_name]
-        op_version = 10
-
-        container.add_node('Resize', inputs, output_name, op_version=op_version, **attrs)
```

Comment:
Is onnxmltools in sync with the latest change of this file?

Reply:
Yes, see: https://github.com/onnx/onnxmltools/pull/279

Every op has been confirmed, and it includes the upsample opset 10 changes.